File: BroadCast/app/src/main/java/com/example/liangzihong/broadcast/Activitys/MainActivity.java
package com.example.liangzihong.broadcast.Activitys;

import android.content.Intent;
import android.content.IntentFilter;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.example.liangzihong.broadcast.BroadcastReceivers.BroadcastReceiver_1;
import com.example.liangzihong.broadcast.R;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    private Button button1;
    private Button button2;
    private static final String MyAction="MyAction";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_layout);
        init();

    }

    private void init() {
        button1=(Button)findViewById(R.id.button1);
        button2=(Button)findViewById(R.id.button2);
        button1.setOnClickListener(this);
        button2.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch(v.getId()){

            //发送标准广播
            //广播的内容是intent
            case R.id.button1:
                sendStandardBroadcast();
                break;
            case R.id.button2:
                mySendOrderedBroadcast();
                break;

            case R.id.button3:
                DynamicReceive();
                break;
        }
    }


    //发送标准广播
    //广播的内容是intent
    private void sendStandardBroadcast() {
        Intent intent1=new Intent();
        intent1.putExtra("name","ali");  //为intent添加信息
        intent1.setAction(MyAction);    //为广播添加密钥，只有符合这个密钥的才能接收
        sendBroadcast(intent1);
    }


    //发送有序广播
    //有序广播可以 被 前面的广播接收者截断或加以处理
    private void mySendOrderedBroadcast() {
        Intent intent2=new Intent();
        intent2.putExtra("name","Wongtsuiyu");  //为intent添加信息
        intent2.setAction(MyAction);    //为广播添加密钥，只有符合这个密钥的才能接收
        sendOrderedBroadcast(intent2,null);

    }

    /**
     * 动态注册
     */
    private void DynamicReceive() {
        IntentFilter intentFilter=new IntentFilter(MyAction);
        BroadcastReceiver_1 bc1=new BroadcastReceiver_1();
        registerReceiver(bc1,intentFilter);
    }

}

File: Dialog/app/src/main/java/com/example/liangzihong/dialog/MainActivity.kt
package com.example.liangzihong.dialog

import android.app.AlertDialog
import android.content.DialogInterface
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private var button1: Button? = null
    private var button2: Button? = null
    private var button3: Button? = null
    private var button4: Button? = null
    private var button5: Button? = null
    private var Dialog_Button: Button? = null
    private var builder: android.app.AlertDialog.Builder? = null
    private val SingleChoice = arrayOf("Teacher", "CTO", "Programmer")
    private val MultiplyChoice = arrayOf("Football", "Basketball", "Tennis")
    private val ItemList = arrayOf("CEO", "CTO", "COO", "Project Manager")
    private var dialog: android.app.AlertDialog? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_layout)

        init()
    }

    /**
     * 按钮的初始化以及事件响应
     */
    private fun init() {
        button1 = findViewById(R.id.button1) as Button
        button2 = findViewById(R.id.button2) as Button
        button3 = findViewById(R.id.button3) as Button
        button4 = findViewById(R.id.button4) as Button
        button5 = findViewById(R.id.button5) as Button

        button1!!.setOnClickListener(this)
        button2!!.setOnClickListener(this)
        button3!!.setOnClickListener(this)
        button4!!.setOnClickListener(this)
        button5!!.setOnClickListener(this)
    }

    /**
     * 触发按钮点击事件
     */
    override fun onClick(v: View) {
        when (v.id) {
        //确认对话框
            R.id.button1 -> {
                builder = AlertDialog.Builder(this)
                //对话框的标题
                builder!!.setTitle("确认对话框")
                builder!!.setIcon(R.mipmap.ic_launcher)

                //对话框的提示内容
                builder!!.setMessage("这是确认对话框的内容")
                //对话框的基本事件响应
                setBuilder(builder)
                dialog = builder!!.create()
                dialog!!.show()
            }


        //单选对话框
            R.id.button2 -> {
                builder = AlertDialog.Builder(this)
                builder!!.setTitle("单选对话框")
                builder!!.setIcon(R.mipmap.ic_launcher)
                //设置单选对话框的数据源以及  对应的事件响应
                builder!!.setSingleChoiceItems(SingleChoice, 0) { dialog, which ->
                    val choose = SingleChoice[which]
                    Toast.makeText(this@MainActivity, "你选择了" + choose, Toast.LENGTH_SHORT).show()
                }

                //对话框的基本事件响应
                setBuilder(builder)
                dialog = builder!!.create()
                dialog!!.show()
            }


        //多选对话框
            R.id.button3 -> {
                builder = AlertDialog.Builder(this)
                builder!!.setTitle("爱好")
                builder!!.setIcon(R.mipmap.ic_launcher_round)
                builder!!.setMultiChoiceItems(MultiplyChoice, null) { dialog, which, isChecked ->
                    if (isChecked) {
                        Toast.makeText(this@MainActivity, "你选择了" + MultiplyChoice[which], Toast.LENGTH_SHORT).show()
                    } else {
                        Toast.makeText(this@MainActivity, "你取消了" + MultiplyChoice[which], Toast.LENGTH_SHORT).show()
                    }
                }
                setBuilder(builder!!)
                dialog = builder!!.create()
                dialog!!.show()
            }

        //列表对话框
            R.id.button4 -> {
                builder = AlertDialog.Builder(this)
                builder!!.setTitle("职位")
                builder!!.setIcon(R.mipmap.ic_launcher_round)

                //设置数据源
                builder!!.setItems(ItemList) { dialog, which -> Toast.makeText(this@MainActivity, "我做了" + ItemList[which], Toast.LENGTH_SHORT).show() }
                setBuilder(builder)
                dialog = builder!!.create()
                dialog!!.show()
            }


        //自定义对话框
            R.id.button5 -> {
                val view = View.inflate(this, R.layout.mydialog_layout, null)

                builder = AlertDialog.Builder(this)
                builder!!.setTitle("自定义对话框")
                builder!!.setIcon(R.mipmap.ic_launcher_round)
                builder!!.setView(view)

                Dialog_Button = view.findViewById(R.id.mydialog_button) as Button
                Dialog_Button!!.setOnClickListener { Toast.makeText(this@MainActivity, "你的名字是", Toast.LENGTH_SHORT).show() }

                setBuilder(builder)
                dialog = builder!!.create()
                dialog!!.show()
            }
        }
    }

    /**
     * 设置对话框的确定与取消
     * @param builder
     */
    private fun setBuilder(builder: AlertDialog.Builder) {
        //对话框的事件响应,按确认和取消都是直接消失
        builder.setPositiveButton("确认") { dialog, which -> dialog.dismiss() }
        builder.setNegativeButton("取消") { dialog, which -> dialog.dismiss() }
    }
}
